tests/unit/telegram_bot/test_webhook_server.py

```
@@ -0,0 +1,94 @@
+            response = await telegram_webhook(mock_request)
+            assert response.status_code == 400
+
+    @pytest.mark.asyncio
+    async def test_unauthorized_returns_401(self):
+        from core.webhook.server import telegram_webhook
+
+        mock_request = MagicMock()
+        mock_request.headers = {"X-Telegram-Bot-Api-Secret-Token": "wrong"}
+
+        with patch("core.webhook.server.settings") as mock_settings:
+            mock_settings.TELEGRAM_BOT_SECRET_TOKEN = "correct"
```

Ruff (0.15.7)

[error] Possible hardcoded password assigned to: "TELEGRAM_BOT_SECRET_TOKEN"

(S105)